tests/Unit/Http/controller/CategoryControllerTest.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Tests\Unit\Http\controller;
+
+use App\Http\Controllers\CategoryController;
+use App\Http\Requests\AddCategoryRequest;
+use App\Models\Category;
+use App\Repositories\Category\CategoryRepositoryInterface;
+use Tests\TestCase;
+use Mockery;
+use Mockery\MockInterface;
+use App\Service;
+use Illuminate\Database\Eloquent\Collection;
+use Illuminate\Http\RedirectResponse;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\View\View;
+
+class CategoryControllerTest extends TestCase
+{
+    protected $controller;
+    protected $repository;
+    protected $formrequest;
+
+    public function setup() : void
+    {
+        parent::setUp();
+        $this->repository =
+            $this->instance(CategoryRepositoryInterface::class, Mockery::mock(CategoryRepositoryInterface::class));
+        $this->controller = new CategoryController($this->repository);
+        $this->formrequest = $this->instance(AddCategoryRequest::class, Mockery::mock(AddCategoryRequest::class));
+    }
+
+    public function tearDown() : void
+    {
+        Mockery::close();
+        parent::tearDown();
+    }
+
+    public function testShowAddCategoryView()
+    {
+        $this->repository->shouldReceive('getAll')->once()
+            ->andReturn(new Collection(array(new Category(), new Category())));
```

Comment:
```suggestion
            ->andReturn(new Collection([new Category(), new Category()]));
```
tương tự bên dưới, ta dùng [] ko dùng hàm array